rln/src/utils.rs

```diff
@@ -1,34 +1,63 @@
 use ark_bn254::{Bn254, Fr, Parameters};
-use ark_ff::{Fp256, PrimeField};
+use ark_ff::{BigInteger, Field as ArkField, FpParameters, PrimeField};
 use ark_std::str::FromStr;
 use ethers_core::utils::keccak256;
 use num_bigint::{BigInt, BigUint, ToBigInt};
 use semaphore::{identity::Identity, Field};
+use std::iter::Extend;
 
-pub fn to_fr(el: Field) -> Fr {
-    Fr::try_from(el).unwrap()
+pub fn modulus_bit_size() -> usize {
+    <Fr as PrimeField>::Params::MODULUS
+        .num_bits()
+        .try_into()
+        .unwrap()
+}
+
+pub fn fr_byte_size() -> usize {
+    let mbs = modulus_bit_size();
+    (mbs + 64 - (mbs % 64)) / 8
+}
+
+pub fn to_fr(el: &Field) -> Fr {
+    Fr::try_from(*el).unwrap()
 }
 
-pub fn to_field(el: Fr) -> Field {
-    el.try_into().unwrap()
+pub fn to_field(el: &Fr) -> Field {
+    (*el).try_into().unwrap()
```

Comment:
Usage of `unwrap` is not recommended in production code, use `Result`. This utility function seems unnecessary.